applications/ConstitutiveLawsApplication/custom_constitutive/auxiliary_files/hcf_data_container.h

```diff
@@ -0,0 +1,313 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alireza Taherzadeh-Fard
+//                   Alejandro Cornejo
+//
+//
+
+# pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_constitutive/auxiliary_files/cl_integrators/high_cycle_fatigue_law_integrator.h"
+#include "custom_utilities/advanced_constitutive_law_utilities.h"
+#include "constitutive_laws_application_variables.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class HCFDataContainer
+ * @ingroup ConstitutiveLawsApplication
+ * @brief Defining all the methods and variables required in fatigue simulations
+ * @details Give access to methods and restores all member variables in fatigue simulations
+ * @author Alireza Taherzadeh-Fard, Alejandro Cornejo
+ */
+class HCFDataContainer
+{
+ConstitutiveLaw::Parameters rValues = ConstitutiveLaw::Parameters();
+public:
+
+    struct FatigueVariables {
```

Comment:
Every variable must be saved in every GP, there are not common values that can be stored in properties?

Reply:
I agree, some `double` values can be recomuted, not being necessary to be stored

Reply:
these are all fatigue variables which have been gathered in one place. Not all of them are stored but the remaining ones are necessary for calculations.